share/completions/john.fish

```diff
@@ -1,28 +1,40 @@
-set -l __fish_john_encodings ASCII RAW UTF-8 ISO-8859-1 Latin1 ANSI ISO-8859-2 ISO-8859-7 ISO-8859-15 KOI8-R CP437 CP720 CP737 CP850 CP852 CP858 CP866 CP868 CP1250 CP1251 CP1252 CP1253 CP1254 CP1255 CP1256
+function __fish_john_encodings --description "Print JohnTheRipper encodings"
+    john --list=encodings 2>/dev/null | string match -a -r '[A-Z][A-Za-z0-9-]*'
+end
 
 function __fish_john_rules --description "Print JohnTheRipper rules"
     john --list=rules 2>/dev/null
 end
 
-function __fish_john_formats --description "Print JohnTheRipper hash formats"
-    john --list=format-details 2>/dev/null | cut -f1
+function __fish_john_formats --description "Complete JohnTheRipper hash formats"
+    john --list=format-details 2>/dev/null | string match -v -r '^dynamic_' | cut -f 1,8
+    john --list=subformats 2>/dev/null | string replace -fr '.*type = (dynamic_\d+): (.*)' '$1\t$2'
 end
 
 complete -c john -l help -d "print usage summary"
 complete -c john -l single -fa "(__fish_stripprefix='^--single=' __fish_complete_list , __fish_john_rules)" -d "single crack mode"
 complete -c john -l single-seed -rf -d "add static seed word(s) for all salts in single mode"
 complete -c john -l single-wordlist -rF -d "short wordlist with static seed words/morphemes"
 complete -c john -l single-user-seed -rF -d "wordlist with seeds per username"
-complete -c john -l single-retest-guess -d "enable SingleRetestGuess"
-complete -c john -l no-single-retest-guess -d "disable SingleRetestGuess"
 complete -c john -l single-pair-max -rf -d "max. number of word pairs generated"
 complete -c john -l no-single-pair -d "disable single word pair generation"
+complete -c john -l single-retest-guess -d "enable SingleRetestGuess"
+complete -c john -l no-single-retest-guess -d "disable SingleRetestGuess"
```

Comment:
> A lot of translated strings were just removed.

if `build_tools/update_translations.fish` causes changes on master,
that's likely a problem on your end,
Do you have GNU gettext installed? This script uses various tools like `msgfmt`, `msguniq`,...

No big deal, I can do it for you (along the other change) if that helps